File: Week 6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters_number = count_letters(text)
    words_number = count_words(text)
    sentences_number = count_sentences(text)

    grade = round(0.0588 * letters_number / words_number * 100 - \
                  0.296 * sentences_number / words_number * 100 - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

def count_letters(text):
    sum = 0
    for i in range(len(text)):
        if text[i].isalpha():
            sum += 1
    return sum

def count_words(text):
    sum = 1
    for i in range(len(text)):
        if text[i].isspace():
            sum += 1
    return sum

def count_sentences(text):
    sum = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sum += 1
    return sum

main()

File: Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int CountLetters(string text);
int CountWords(string text);
int CountSentences(string text);

int main(void)
{

    string text = get_string("Text: ");
    int letters_number = CountLetters(text);
    int words_number = CountWords(text);
    int sentences_number = CountSentences(text);

    float grade = 0.0588 * letters_number / (float)words_number * 100 - 0.296 * sentences_number / (float)words_number * 100 - 15.8;

    int i_grade = round(grade);

    if (i_grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (i_grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", i_grade);
    }

}

int CountLetters(string t)
{
    int sum = 0;

    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if (isalpha(t[i]))
        {
            sum++;
        }
    }
    return sum;
}

int CountWords(string t)
{
    int sum = 1;

    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if (isblank(t[i]))
        {
            sum++;
        }
    }
    return sum;
}

int CountSentences(string t)
{
    int sum = 0;

    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if (t[i] == '.' || t[i] == '?' || t[i] == '!')
        {
            sum++;
        }
    }
    return sum;
}
